### PR TITLE
Merge 1.7.6.x to develop - 16/04/2019

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Daily merge of bug fixes from `1.7.6.x` branch to `develop` as we are in code freeze
| Type?         | bug fix
| Category?     | ME
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Travis should be green

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13430)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/pages/categories/index.js [hunk 1]
```diff
@@ -34,8 +34,10 @@ import SubmitRowActionExtension from "../../components/grid/extension/action/row
 import LinkRowActionExtension from "../../components/grid/extension/link-row-action-extension";
 import CategoryPositionExtension from "../../components/grid/extension/column/catalog/category-position-extension";
 import AsyncToggleColumnExtension from "../../components/grid/extension/column/common/async-toggle-column-extension";
-import DeleteCategoryRowActionExtension from "../../components/grid/extension/action/row/category/delete-category-row-action-extension";
-import DeleteCategoriesBulkActionExtension from "../../components/grid/extension/action/bulk/category/delete-categories-bulk-action-extension";
+import DeleteCategoryRowActionExtension
+  from "../../components/grid/extension/action/row/category/delete-category-row-action-extension";
```
Comment (Quetzacoalt91) [inline admin-dev/themes/new-theme/js/pages/categories/index.js:38]: Is this code valid?

Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/pages/categories/index.js:38]: Yep, but I don't know why they add this in two lines. and why the path is sooooo lonng :D

**Review by Quetzacoalt91 — APPROVED** [review]

File: admin-dev/themes/new-theme/js/pages/categories/index.js (hunk 1 above)
Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/pages/categories/index.js:38]: `extension/action/row/category/delete-category-row-action-extension` so redundant :/
we repeat the path in the name of the file... `extension/action/row/category/delete.js` is enough
ping @matks

Comment (matks) [reply admin-dev/themes/new-theme/js/pages/categories/index.js:38]: If you mean the `import` statement splitted in 2 lines, yes. I'm not a fan too but this is suggested by phpstorm linter and helps readability so ... I went for it 👐

Comment (matks) [reply admin-dev/themes/new-theme/js/pages/categories/index.js:38]: > `extension/action/row/category/delete-category-row-action-extension` so redundant :/
> we repeat the path in the name of the file... `extension/action/row/category/delete.js` is enough

This is an old debate 😛 "should the name of a class carry some information if the path already contains it ?"

Pros:
- avoids generic names confusion, useful when searching for a class by name in the codebase
- when class is used out of context, name is self-explanatory so unambiguous

Cons:
- redundant information
- longer class names

I think both sides are OK, tell me if you'd want some cleanup and to set a global rule here

Comment (Quetzacoalt91) [reply admin-dev/themes/new-theme/js/pages/categories/index.js:38]: Is this something to handle in this PR?

I was going to merge it, but as long as this discussion is in progress we have to delay it.

Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/pages/categories/index.js:38]: >avoids generic names confusion, useful when searching for a class by name in the codebase
>when class is used out of context, name is self-explanatory so unambiguous

This is why we invented namespaces 😉

Comment (matks) [reply admin-dev/themes/new-theme/js/pages/categories/index.js:38]: > Is this something to handle in this PR?
> 
> I was going to merge it, but as long as this discussion is in progress we have to delay it.

As this is a PR to merge 1.7.6.x in develop, I suggest we fix ir in a dedicated issue in 1.7.6.x or develop

Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/pages/categories/index.js:38]: >avoids generic names confusion, useful when searching for a class by name in the codebase
>when class is used out of context, name is self-explanatory so unambiguous

This is why we invented namespaces 😉

**Comment by PierreRambaud** [comment]
Thanks @matks

**Pull request merged**